File: anagram.java
import java.io.*;
import java.util.*;
public class anagram
{
	public static String remove_sc(String str)
	{
		char ch[]=str.toCharArray();
		int len=ch.length;
		int j=0,k;
		char ch1[]=new char[30];
		for(int i=0;i<len;i++)
		{
			k=ch[i];
			if((k>=65 && k<=90) || (k>=97 && k<=122))
				ch1[j++]=ch[i];
		}
		String str1=new String(ch1);
		return str1;
	} 
	public static char anag(String str,String str1)
	{
		String str2=str.toLowerCase();
		String str3=str1.toLowerCase();
		char ch[]=str2.toCharArray();
		char ch1[]=str3.toCharArray();
		int len1=ch.length;
		int len2=ch1.length;
		int cnt=0,cnt1=0,asc,asc1;
		if(len1==len2)
		{
			for(int i=0;i<len1;i++)
			{
				for(int j=0;j<len1;j++)
				{
					asc=ch1[i];
					asc1=ch[j];
					if(asc==asc1)
					{
						cnt++;
						break;
					}
				}
			}
			for(int i=0;i<len1;i++)
			{
				for(int j=0;j<len1;j++)
				{
					asc=ch[i];
					asc1=ch1[j];
					if(asc==asc1)
					{
						cnt1++;
						break;
					}
				}
			}
			if(cnt==len1 && cnt1==len2)
				return 'y';
			else
				return 'n';
		}
		else 
			return 'n';
	}
	public static void main(String[] args) throws IOException
	{
		FileReader fr=new FileReader("sample1.txt");
		BufferedReader br=new BufferedReader(fr);
		String l;
		String str;
		String str1;
		String s[]=new String[1000];
		String s1[][]=new String[100][100];
		String ws[]=new String[1000];
		int num[]=new int[100];
		int now=0,cnt=0,m=0;
		while((l=br.readLine())!=null)
		{
			StringTokenizer st=new StringTokenizer(l," ");
			while(st.hasMoreTokens())
			{
				str=st.nextToken();
				str1=remove_sc(str);
				s[now++]=str1.trim();
			}
		}
		for(int i=0;i<now;i++)
		{
			int k;
			if(i!=0)
			{
			for(k=0;k<cnt;k++)
			{
				if(num[k]==i)
					break;
			}
			if(num[k]==i)
				continue;
			}
			k=0;
			s1[m][k++]=s[i];
			for(int j=i+1;j<now;j++)
			{
				char ch=anag(s[i],s[j]);
				if(ch=='y')
				{
					char ch1='n';
					for(int n=0;n<k;n++)
					{
						if(s[j].equals(s1[m][n]))
						{
							ch1='y';
							num[cnt++]=j;
						}
					}
					if(ch1=='n')
					{
						s1[m][k++]=s[j];
						num[cnt++]=j;
					}
				}
			}
			m++;
		}
		for(int i=0;i<m;i++)
		{
			int cnt1=0;
			for(int j=0;j<s1[m].length;j++)
			{
				if(s1[i][j]!=null)
					cnt1++;
			}
			if(cnt1>=2)
			{
				int j;
				for(j=0;j<cnt1-1;j++)
				{
					System.out.print(s1[i][j]+",");
				}
				System.out.println(s1[i][j]);
			}
		}
	}	
}

File: anagram.c
#include <stdio.h>


typedef struct node
{
	char s[10][10];
	int len,cnt;
	struct node *next;
}node;

node *create_new();

node *start,*newnode,*ptr,*prev;
char anag(char[],char[]);

void main()
{
	FILE *fp;
	char ch,c,ch1='n',str[10];
	int asc,i=0,j;
	fp=fopen("sample1.txt","r");
	while((c=getc(fp))!=EOF)
	{
		asc=c;
		if((asc>=65 && asc<=90) || (asc>=97 && asc<=122))
		{
			str[i++]=c;
		}
		else
		{
			str[i]='\0';
			if(start==NULL)
			{
				start=create_new();
				for(j=0;j<i;j++)
					start->s[start->cnt][j]=str[j];
				start->s[start->cnt][j]='\0';
				start->len=i;
				start->cnt=start->cnt+1;
			}
			else
			{
				for(ptr=start;ptr;prev=ptr,ptr=ptr->next)
				{
				       if(ptr->len==i)
				       {
						for(j=0;j<i;j++)
							ptr->s[ptr->cnt][j]=str[j];
						ptr->s[ptr->cnt][j]='\0';
						ptr->cnt=ptr->cnt+1;
						break;
				       }
				}
				if(ptr!=NULL)
				{
					i=0;
					continue;
				}
				else
				{
					ptr=create_new();
					prev->next=ptr;
					for(j=0;j<i;j++)
						ptr->s[ptr->cnt][j]=str[j];
					ptr->s[ptr->cnt][j]='\0';
					ptr->len=i;
					ptr->cnt=ptr->cnt+1;
					i=0;
				}
			}
		}
	}
	for(ptr=start;ptr;ptr=ptr->next)
	{
		if(ptr->len>=4)
		{
			for(i=0;i<ptr->cnt;i++)
			{
				printf("%s",ptr->s[i]);
				for(j=i+1;j<ptr->cnt;j++)
				{
					ch1=anag(ptr->s[i],ptr->s[j]);
					if(ch1=='y')
					{
						printf("%s,%s ",ptr->s[i],ptr->s[j]);
						ch1='n';
					}
				}
			}
		}
	}
}

node *create_new()
{
	newnode=(node*)malloc(sizeof(node));
	newnode->next=NULL;
	newnode->len=0;
	newnode->cnt=0;
	return newnode;
}

char anag(char s[10],char s1[10])
{
	int i,j,k,len2,len1,cnt1=0,cnt2=0,asc,asc1;
	for(i=0;s[i]!='\0';i++);
	len1=i;
	for(i=0;s1[i]!='\0';i++);
	len2=i;
	if(len1==len2)
	{
		for(i=0;i<len1;i++)
		{
			for(j=0;j<len1;j++)
			{
				asc=s[i];
				asc1=s1[j];
				if(asc==asc1 || asc==asc1-32)
				{
					cnt1++;
					break;
				}
			}
		}
		for(i=0;i<len1;i++)
		{
			for(j=0;j<len1;j++)
			{
				asc=s1[i];
				asc1=s[j];
				if(asc==asc1 || asc==asc1-32)
				{
					cnt2++;
					break;
				}
			}
		}
		if(cnt1==len1 && cnt2==len2)
			return 'y';
		else
			return 'n';
	}
	else
		return 'n';
}
